cpp/rss_subset_aggregator.cc

```diff
@@ -0,0 +1,315 @@
+/** \file   rss_subset_aggregator.cc
+ *  \brief  Aggregates RSS feeds.
+ *  \author Dr. Johannes Ruscheinski (johannes.ruscheinski@uni-tuebingen.de)
+ *
+ *  \copyright 2021 Universitätsbibliothek Tübingen.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <algorithm>
+#include <map>
+#include <unordered_map>
+#include <unordered_set>
+#include <cinttypes>
+#include <cstring>
+#include "Compiler.h"
+#include "DbConnection.h"
+#include "DbResultSet.h"
+#include "EmailSender.h"
+#include "FileUtil.h"
+#include "HtmlUtil.h"
+#include "SqlUtil.h"
+#include "StringUtil.h"
+#include "SyndicationFormat.h"
+#include "Template.h"
+#include "UBTools.h"
+#include "util.h"
+#include "VuFind.h"
+#include "XmlWriter.h"
+
+
+namespace {
+
+
+[[noreturn]] void Usage() {
+    ::Usage("--mode=(email|rss_xml) (user_id|error_email_address) subsystem_type]\n"
```

Comment:
1) A user id is unique for all subsystem, so it shouldn't be necessary to add a subsystem_type if a user_id is given (only if --mode=email)
2) If i call it right now with --mode=rss_xml, the following messages are shown:

```
free(): double free detected in tcache 2
Abgebrochen (Speicherabzug geschrieben)
```